src/include/miopen/readonlyramdb.hpp

```diff
@@ -106,6 +105,46 @@ class ReadonlyRamDb
     void Prefetch(const std::string& path, bool warn_if_unreadable);
 };
 
+struct FindRamDb : ReadonlyRamDb
+{
+    const std::unordered_map<std::string, std::string>& find_db_init(std::string arch_cu);
+    FindRamDb(std::string path, std::string _arch, std::size_t _num_cu)
+        : ReadonlyRamDb(":memory:" + path, _arch, _num_cu)
+    {
+        static const auto m = find_db_init(arch + "_" + std::to_string(num_cu));
```

Comment:
By the way, you have 3 in-memory copies of each map. You should either std::move them (which would be simplest) or use pointers/refs.
Copies:

1. Static variable in the generated file.
2. This static variable.
3. Field of the db instance.

Reply:
+ 4th ~copy is generated file variable initializer.

Reply:
Nice catch! Fixed

Reply:
So. Now it has two in-memory direct 100% copies of the map and one initializer, which is de facto at least a copy of map content.